File: LPM.Database/Models/OrderAppointment.cs
using LPM.Database.Enums;
using LPM.Database.Models.Base;

namespace LPM.Database.Models
{
    /// <summary>
    /// Приказ о назначении
    /// </summary>
    public class OrderAppointment : BaseEntity<Guid>
    {
        /// <summary>
        /// ID сотрудника
        /// </summary>
        public Guid EmployeeId { get; set; }

        /// <summary>
        /// Сотрудник
        /// </summary>
        public Employee Employee { get; set; }

        /// <summary>
        /// ID департамента
        /// </summary>
        public Guid DepartmentId { get; set; }

        /// <summary>
        /// Департамент
        /// </summary>
        public Department Department { get; set; }

        /// <summary>
        /// Дата начала работы
        /// </summary>
        public DateTime DateStart { get; set; }

        /// <summary>
        /// Дата трудоустройства
        /// </summary>
        public DateTime OficialDateStart { get; set; }

        /// <summary>
        /// Дата окончания
        /// </summary>
        public DateTime? DateEnd { get; set; }

        /// <summary>
        /// Должность
        /// </summary>
        public string Position { get; set; }

        /// <summary>
        /// Тип позиции
        /// </summary>
        public EmployeeType EmployeeType { get; set; }
    }
}

File: LPM.WebApp/src/app/modules/personnel-management/stuff/orderAppointment/orderAppointment.ts
import { Guid } from 'guid-typescript';
import { BaseEntity } from 'src/app/modules/shared/models/baseEntity';
import { SelectItem } from 'src/app/modules/shared/models/selectItem';

export class OrderAppointment extends BaseEntity {
	employeeId: Guid;

	/* Отдел */
	organization: SelectItem;

	/* Отдел */
	department: SelectItem;

	/* Дата начала работы фактическая */
	dateStart: Date;

	/* Дата начала работы официально */
	oficialDateStart?: Date;

	/* Дата окончания испытательного срока */
	probationEndDate?: Date;

	/* Дата окончания исполнения обязанностей по договору */
	dateEnd?: Date;

	/* Должность */
	position?: string;

	/* Тип сотрудника */
	employeeType: SelectItem;

	/* Стаж */
	workLength: number;
}